remote/internal/metrics.go

```diff
@@ -33,38 +43,42 @@ type PendulumEntry struct {
 //
 // Returns:
 // - A slice of PendulumMetric structs containing the aggregated metrics.
-func AggregatePendulumMetrics(data [][]string, timeout_len float64, rangeType string) []PendulumMetric {
-	// get number of columns
-	n := len(data[0])
-
+func AggregatePendulumMetrics(
+	data [][]string,
+	timeout_len float64,
+	rangeType string,
+	sections []interface{}) []PendulumMetric {
 	// create waitgroup
 	var wg sync.WaitGroup
 
 	// create buffered channel to store results and avoid deadlock in main
-	res := make(chan PendulumMetric, n-2)
+	res := make(chan PendulumMetric, len(sections))
```

Comment:
This also likely should remain the same as before to avoid creating entries for the 'time' and 'active' columns since they will never be included in the report.

Reply:
You had it  as `n-2`  as it excludes `active` and `times`. 

Although, as we allows more exclusion, this value will have to vary, won't it?

Reply:
yeah I think you are right actually, I missed that it was using sections and not data.